firebase-perf/e2e-app/src/androidTest/java/com/google/firebase/testing/fireperf/FirebasePerformanceScreenTracesTest.java

```diff
@@ -49,7 +51,11 @@ public void scrollRecyclerViewToEnd() {
       onView(withId(R.id.rv_numbers)).perform(scrollToPosition(currItemCount));
       currItemCount += 5;
     }
-    // End Activity screen trace by switching to another Activity
-    scenario.launch(FirebasePerfScreenTracesActivity.class);
+    assertThat(scenario.getState()).isEqualTo(State.RESUMED);
+    scenario.moveToState(State.STARTED).moveToState(State.CREATED); // trigger activity screen trace
+    // Wait for TransportManager and Firelog executors to finish
+    Thread.sleep(5000);
+    // Block until all Fireperf events are sent by Firelog
+    InstrumentationTestUtil.flgForceUploadSync();
```

Comment:
Same as above, should we cause another activity state transition to make sure we are dispatching events?